runtime/Java/src/org/antlr/v4/runtime/ParserRuleContext.java

```diff
@@ -101,9 +102,8 @@ public void copyFrom(ParserRuleContext ctx) {
 			this.children = new ArrayList<>();
 			// reset parent pointer for any error nodes
 			for (ParseTree child : ctx.children) {
-				if ( child instanceof ErrorNodeImpl ) {
-					this.children.add(child);
-					((ErrorNodeImpl) child).parent = this;
+				if ( child instanceof ErrorNode ) {
+					addChild((ErrorNode)child);
```

Comment:
The node copied from is candidate for GC as all refs to it disappear; figured it's a waste to alter it when it's dead.  Maybe a better comment? :)